src/github.com/matrix-org/dendrite/clientapi/routing/login.go

```diff
@@ -28,6 +29,14 @@ import (
 	"github.com/matrix-org/util"
 )
 
+type loginType string
+
+// https://matrix.org/docs/spec/client_server/r0.3.0.html#login
+const (
+	PasswordBased loginType = "m.login.password"
```

Comment:
Just a heads up that we already have the known logintypes defined [here](https://github.com/matrix-org/dendrite/blob/master/src/github.com/matrix-org/dendrite/clientapi/auth/authtypes/logintypes.go).